test/shared_list.rb

```diff
@@ -105,18 +113,24 @@ def test_insert_at
       new4.reload
       assert_equal 5, new4.pos
 
-      last1 = ListMixin.order('pos').last
-      last2 = ListMixin.order('pos').last
+      new_noup.reload
+      assert_equal $default_position, new_noup.pos
+
+      last1 = ListMixin.where('pos IS NOT NULL').order('pos').last
+      last2 = ListMixin.where('pos IS NOT NULL').order('pos').last
       last1.insert_at(1)
       last2.insert_at(1)
-      assert_equal [1, 2, 3, 4, 5], ListMixin.where(parent_id: 20).order('pos').map(&:pos)
+      if ENV['DB'] == 'postgresql' and $default_position.nil?
```

Comment:
You could use this: http://stackoverflow.com/a/20959470/129798

So instead of changing what's expected, augment the scope conditionally:

`ListMixin.where(parent_id: 20).order('pos NULLS FIRST').map(&:pos)`